tests/test_move.py

```diff
@@ -60,3 +63,91 @@ def test_move_dir(self):
         if self.preserve_time:
             dst_file2_info = dst_fs.getinfo("bar/baz.txt", namespaces)
             self.assertEqual(dst_file2_info.modified, src_file2_info.modified)
+
+
+class TestMove(unittest.TestCase):
+    def test_move_file(self):
+        with open_fs("temp://") as temp:
+            syspath = temp.getsyspath("/")
+            a = open_fs(syspath)
+            a.makedir("dir")
+            b = open_fs(join(syspath, "dir"))
```

Comment:
IMHO this looks a bit weird (creating a file in the subdirectory of a syspath of an already-existing FS).
I think your optimised code-path would still get followed if `a` and `b` were both independent TempFS instances?